controllers/nnf_workflow_controller_helpers.go

```diff
@@ -849,3 +1002,484 @@ func (r *NnfWorkflowReconciler) removeAllPersistentStorageReferences(ctx context
 
 	return nil
 }
+
+func (r *NnfWorkflowReconciler) createContainerService(ctx context.Context, workflow *dwsv1alpha1.Workflow) error {
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      workflow.Name,
```

Comment:
If there is more than one "#dw container" line, is it okay that you're only creating one service?

Reply:
Great question. I haven't test multiple container directives, which is something I will do in the future. Just having 1 service would allow all containers to talk to each other if need be. If we need to further partition this by the directive index and create multiple services, I think we could do that.

The service name ends up being the subdomain for the container (i.e. hostname.subdomain.namespace.svc.clusterlocal), so that could get a bit confusing.

Reply:
I put in an a check in proposal for multiple container directives. For now, it will error out. The way things are currently working, if you supply multiple container directives you only get 1 job as well since it's based on the workflow name.